From 1772d4291a382f48ac8aa8afa67beb94f14c2a82 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Fri, 13 Oct 2017 13:05:47 +0800
Subject: [PATCH 2644/5242] MLK-16581-5 gpu: imx: dpu: common: Add helper
 dpu_intsteer_enable_lines() support

commit  3d5424a0b5d3b4acbd7ec25496a48d66396bad2d from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch adds helper dpu_intsteer_enable_lines() support so that
users may enable intsteer lines with one function call.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-common.c |   18 +++++++++++++-----
 1 file changed, 13 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/imx/dpu/dpu-common.c b/drivers/gpu/imx/dpu/dpu-common.c
index 84c7bee..ff8dd52 100644
--- a/drivers/gpu/imx/dpu/dpu-common.c
+++ b/drivers/gpu/imx/dpu/dpu-common.c
@@ -1099,13 +1099,10 @@ static void dpu_intsteer_enable_line(struct dpu_soc *dpu, unsigned int line)
 			   BIT(shift), BIT(shift));
 }
 
-static int dpu_irq_init(struct dpu_soc *dpu)
+static void dpu_intsteer_enable_lines(struct dpu_soc *dpu)
 {
 	const struct dpu_devtype *devtype = dpu->devtype;
-	const struct cm_reg_ofs *ofs = devtype->cm_reg_ofs;
-	struct irq_chip_generic *gc;
-	struct irq_chip_type *ct;
-	int ret, i;
+	int i;
 
 	for (i = 0; i < devtype->intsteer_map_size; i++) {
 		if (devtype->intsteer_map[i] == NA)
@@ -1113,6 +1110,17 @@ static int dpu_irq_init(struct dpu_soc *dpu)
 
 		dpu_intsteer_enable_line(dpu, devtype->intsteer_map[i]);
 	}
+}
+
+static int dpu_irq_init(struct dpu_soc *dpu)
+{
+	const struct dpu_devtype *devtype = dpu->devtype;
+	const struct cm_reg_ofs *ofs = devtype->cm_reg_ofs;
+	struct irq_chip_generic *gc;
+	struct irq_chip_type *ct;
+	int ret, i;
+
+	dpu_intsteer_enable_lines(dpu);
 
 	dpu->domain = irq_domain_add_linear(dpu->dev->of_node,
 					    devtype->intsteer_map_size,
-- 
1.7.9.5

